### ISSUE TITLE
Line numbers might be incorrect in LiquidSyntaxError for multi-line templates

### ISSUE THREAD

**Issue by nasudadada** [jg-rp/liquid#162]
Hi there!

I think I've found an issue with line number reporting in error messages, so I wanted to check if this is expected behavior.

### Description
I noticed that in some cases with multi-line templates, the line number in LiquidSyntaxError messages might not be showing the actual line where the error occurred.

### Steps to Reproduce
```python
from liquid import Template

template_string = """Hello, 
My name is {{{ name }}
"""

try:
    Template(template_string)
except LiquidSyntaxError as e:
    print(f"Error message: {e}")
```

### Expected Behavior
I would expect the error message to show line 2, as that's where the syntax error appears to be.

### Actual Behavior
The error message seems to always show line 1 in this case.

### Investigation
I've been looking at the error handling in `environment.py`, and I wonder if this part might be related:

```python
if not isinstance(exc, Error):
    exc = exc(msg, linenum=linenum)
elif not exc.linenum:
    exc.linenum = linenum
```

### Questions
1. Is this the intended behavior?
2. If not, would it make sense to update the line number handling?
3. Are there any test cases that specifically check for correct line numbers in syntax errors?

### Additional Context
- I noticed that some test cases in `test_malformed.py` do handle line numbers correctly, so I might be missing something about how this should work.

I'd be happy to investigate further or help with a fix if this turns out to be a bug. Please let me know if you'd like me to provide any additional information or create a PR.

Thank you for maintaining this great library!

**Comment by jg-rp** [comment]
Hi @nasudadada,

Thanks for reporting this. It is not intended behaviour. I was failing to pass a line number to some expression parsing functions in multiple place, causing the line number to default to `1`.

I still needs a few more tests, but PR #163 should fix this.

**Comment by nasudadada** [comment]
Hi @jg-rp 

Thank you for the quick response and for already working on the fix! 
Really appreciate your prompt attention to this issue.

**Comment by jg-rp** [comment]
This should be fixed in version 1.12.2, now released.

I've been working on improving exceptions and error messages in [Python Liquid2](https://github.com/jg-rp/python-liquid2) (separate repository). Using the example above, we get the following error message from Python Liquid2.

```python
from liquid2 import render

source = """\
Hello,
My name is {{{ name }}
"""

render(source, name="Bob")
```

```
liquid2.exceptions.LiquidSyntaxError: unexpected '{'
  -> 'My name is {{{ name }}' 2:13
  |
2 | My name is {{{ name }}
  |              ^ unexpected '{'

```

**Comment by nasudadada** [comment]
Thank you so much for the quick fix and release of version 1.12.2! 

I've confirmed that the issue has been resolved in the new version.

Thanks for maintaining such a great library!
